### ISSUE TITLE
Infographic template for population and minimum needs

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#3473]
## Problem
We need an infographic template for population and minimum needs report. We have a good report on a map and table result from InaSAFE analysis, in the future we must try in an easy way to represent the data to the infographic style. I have written a RFC about infographic #3206 but in the short term maybe is a good time to introduce in a simple report to show the impact analysis of population and minimum needs into infographic report. 

## Proposed solution
Make an infographic template for population and minimum needs reports. The colour of the graph based on https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md 
The icon to support the template based on #3366 .  And about the graphic elements based on #3472 . I collect and zip the icon (in SVG format files) and a note inside the zip file about the icon.

## Source of an icon
This an icon in SVG format files:
- Population icon: [People SVG.zip](https://github.com/inasafe/inasafe/files/610812/People.SVG.zip)
- Minimum needs icon: [Minimum needs SVG.zip](https://github.com/inasafe/inasafe/files/610813/Minimum.needs.SVG.zip)

## Cc
@Charlotte-Morgan @timlinux @lucernae @Gustry @ismailsunni @fredychandra @samnawi @ivanbusthomi

**Comment by Charlotte-Morgan** [comment]
Hi @felix-yew - thanks for updating this ticket.
Can you please put the details about icons in the icons ticket for population and min needs #3360 
This ticket is about the template - ie page layout can you please put your templates in here?

**Comment by felix-yew** [comment]
Hi, @Charlotte-Morgan - about the template for population and minimum needs, I have an example of an infographic. First, we only represent the population and minimum needs data from InaSAFE analysis so it will be great if we can represent and print it into the fit size paper, like in A5 size (148 x 210 mm). I have an example of the pattern to show or represent the population and minimum needs analysis reports. The example comes from https://contently.com/strategist/2012/02/14/warby-parker-annual-report-2011/ 
![warby-parker-annual-report-company-culture](https://cloud.githubusercontent.com/assets/13621886/20998074/b204f4a8-bd3d-11e6-99a4-7b5473d0db8e.png)

As the example, I choose the infographic because I pick up the pattern of represent the data. The different from the example, I change the text parameter into icons. I create a template and adapt the pattern to represent our data. 
![basic template](https://cloud.githubusercontent.com/assets/13621886/21001015/95e9829e-bd50-11e6-8de2-c104b6e0a0fa.jpg)
From the template we can make :

- The title refers to title in inasafe report, but it will be better if we can add the object name (place or district). And, the inasafe logo used in white color and add a little infographic text below the inasafe logo.
- The doughnut chart will represent the people affected in each impacted hazard area. The impacted color will refers to reporting standards of impacted color. And will add the number in the centre in each part in this chart. Also, will put the percentage of people affected in each impacted hazard area. In the middle of chart will appear the total of population in the area analysis and in soft color.
- In population area, will represent the people and vulnerability affected. In the people affected only represent the number of affected. And, in the vulnerability affected will represent the number of affected and percentage from people affected.
- In minimum needs area, will represent the minimum needs with icons and the number of total needs in each unit.
- In marginalia, will represent the inasafe version analysis, time of analysis, contact info from inasafe website, and inasafe supporters in text.

And, this is a mock up of infographic reports :
![people and minimum needs 2](https://cloud.githubusercontent.com/assets/13621886/21000956/3f9106d8-bd50-11e6-8166-35dc4ae9b964.jpeg)



Happy to discuss it :smile:

**Comment by felix-yew** [comment]
Another mock up of infographic report:
![people and minimum needs 3](https://cloud.githubusercontent.com/assets/13621886/21305140/5865f2b8-c5fc-11e6-9bc2-0f43d53e7f83.png)

Or,
![people and minimum needs 4](https://cloud.githubusercontent.com/assets/13621886/21305164/7ec8bd32-c5fc-11e6-910d-466687258240.png)

Consider with another template:

- The number of people and vulnerability affected use rounding.
- Use divider to represent the information.
- Use the scalable icon for the youth,

**Comment by timlinux** [comment]
Just a note to say this is on our radar @felix-yew - Rizky is going to respond with more detailed feedback. I would just like to clarify how you envisage the workflow here: Do you intend the user should see this in the dock widget (I would very much like this)? If se we also need to consider that we will need a vertical layout version of this. I asked @lucernae to use some responsive styling so that it switches between horizontal and vertical layout automatically, but please confirm for us your expectations. Thanks!

**Comment by felix-yew** [comment]
Hi, @timlinux - thank you for your feedback.

That is a great idea if the user can see the infographic in the dock widget, although the user can also print the result, too?. I will try to create a vertical layout. After I finish the vertical template, I will share it again.

**Comment by Charlotte-Morgan** [comment]
Hi @timlinux and @felix-yew - the intention here was to produce a template for a page layout in the first instance to 1) test if its possible to make the graphic elements - @lucernae to provide advice; and 2) as a demonstration of new functionality to be released in v4.0 in January.

@felix-yew, based on the purpose of this work, making a vertical template is not a priority.

**Comment by Charlotte-Morgan** [comment]
@felix-yew  - I really like the changes you made to the template to use panels to highlight elements and to change the emphasis of the icons and text. - I think it looks good.  Can you please make sure that the icons and sources are all documented for this one example so that we can add them to the code?

**Comment by felix-yew** [comment]
Hi, @Charlotte-Morgan - 

> Can you please make sure that the icons and sources are all documented for this one example so that we can add them to the code?

The icons support already created in #3366 . I only modify for the adult's icon, unify the icons of man and woman (Source: UNOCHA) and the icons are in the same tickets #3366 .

**Comment by Charlotte-Morgan** [comment]
Hey @felix-yew  - that is great. What about the 'default size' of icons - did you make some smaller

**Comment by timlinux** [comment]
@Charlotte-Morgan ok sounds good - @lucernae take note of her comments above and focus only on the print layout for now.

**Comment by Charlotte-Morgan** [comment]
@timlinux @lucernae  - i was not expecting this to be implemented till 4.0 release.
ie none of the work to load the icons or document them has been done yet

**Comment by lucernae** [comment]
@felix-yew so, what @timlinux meant by print layout is basically we might want to support different layout for the infographic. For example, this is a landscape version of the report. But in InaSAFE's dock, the report needs to be in portrait orientation if we want to show it in the dock. It is just about the arrangement of the elements. Something like **Responsive Web Design**, it doesn't have to change the whole layout, but only the position of the elements if the size is in portrait orientations.

For example, what can I think of right now is showing the pie chart, then below, the people affected, then below, the vulnerability, then below, the minimum needs (perhaps arranged in two rows).

**Comment by Charlotte-Morgan** [comment]
hi @lucernae, @felix-yew @timlinux - we need to be clear on our intention here. Maybe I can explain.
Imagine you are walking though a market and someone offers you a taste of the latest tasty snacks ... mmmm ... yum. You like it and you decide you want to buy snacks to share with your friends; but the company is not ready to go into production, they are just testing the market. They have many snack options and are deciding which one to make into production.  You have to wait to be able to buy the bulk pack to share.
It's the same here, the info graphic report for min needs is a sample. We think info graphics are a good way to communicate results #3206 and we look for suitable icons #3360 ; but we need to test a sample to see what is possible. We make min needs the test case (we guess this is the most useful / tasty) so we make a subset of icons just for this specific example #3366  and we make a template to demonstrate the use case #3473 .. Thanks to @felix-yew for all this work.
But - we also need to prove it is possible before we put in too much effort - See #3472 
and - we need to load all the icons into the code and document them before we can produce the sample - see #3360 
So - back to the snacks. The min needs icons and template are a taste test. We are not ready to go into production until we have #3472 confirmed and the icons loaded.
For now the intention is to make a A4 landscape pdf page layout for the sample.
If this works we should consider how else to use info graphics - do we want to go to production when our sample is limited and not comprehensive,  should they be used in the dock, what about the web interface we have planned??
I think we should focus on  the sample now as a demonstration / taste test for what is possible in the future. 
@felix-yew - please no more work on this until we get an update on what is possible from @lucernae 
@timlinux  - lets discuss resources required to progress this line of work

**Comment by Charlotte-Morgan** [comment]
@lucernae - I think we would like some assurance from you that the elements proposed in the RFC and tickets are possible - do you have feedback for us?
`It is just about the arrangement of the elements. `

**Comment by timlinux** [comment]
@Charlotte-Morgan understood:

* we are going to focus on producing an A4 Landscape only for 4.0 final

**Comment by Charlotte-Morgan** [comment]
hi @felix-yew - I am looking at the population icons and I do not have the 'people icon" in the #3366 files

**Comment by lucernae** [comment]
Just to tag the picture for the blue version

![Blue Template](https://cloud.githubusercontent.com/assets/831865/22251690/460c6324-e27e-11e6-969d-60850a0a33c1.png)

**Comment by timlinux** [comment]
@lucernae is that a picture of what you producing, or just the blue version of the mockup that @felix-yew made?

**Comment by lucernae** [comment]
from @felix-yew. I put it here so it doesn't get lost in the original PR>

**Comment by lucernae** [comment]
I just realized that for infographics (for donut charts), does it still makes sense/consistent?

- Donut chart is showing affected population per hazard class
- Total population is showing total affected population (or displaced?)
- Vulnerabilities is showing based on displaced population
- Minimum needs is showing based on displaced population

Should I go on with this or is there some revision?

@Charlotte-Morgan @timlinux @felix-yew what do you think?

**Comment by Charlotte-Morgan** [comment]
hi @lucernae - I think this is all fine with discussion in #3634 

The only change i would make to your list above it the total population - I think it should be for the analysis area.
-  Donut for affected by class (where affected is true)
- Total population can show total population in the analysis area 
- Vulnerabilities  are based on displaced population
- Minimum needs are based on displaced population

**Comment by lucernae** [comment]
Ok, thanks!

**Comment by Charlotte-Morgan** [comment]
I hope that is easy :)

**Comment by timlinux** [comment]
@lucernae can you post a screenie of how the infographics you are working on look from time to time so we don't get a huge surprise when you are done?

**Comment by lucernae** [comment]
Sure.
Note that:
- donut chart looks funny because there is still an issue with the affected calculations (total affected is not consistent)
- vulnerabilities not rendered because gender/age ratios not set in aggregations
- all the icons now in png, instead svg. Since, even if the format is svg (from what @felix-yew gave us), it is in fact a png inside. So, it is better to use png again.

<img width="1175" alt="screen shot 2017-01-25 at 20 26 09" src="https://cloud.githubusercontent.com/assets/831865/22292165/9182412a-e33c-11e6-99bc-73800b0cbf2c.png">

**Comment by Gustry** [comment]
> total affected is not consistent)
Is-it the issue that you gave me a long time ago ?
or is it since the new definition ? (total = total affected + total not affected + total not exposed)

**Comment by timlinux** [comment]
Thanks @lucernae that is looking good .... and I love the funky donut :-)

> all the icons now in png, instead svg. Since, even if the format is svg (from what @felix-yew gave us), it is in fact a png inside. So, it is better to use png again.

Yes I think we should ask Maria to fix this (pngs embedded in SVGs) - we are going to get very fuzzy looking images if we scale them up and it breaks the benefit we get from using SVG. I think it is fine as a general approach to rasterize the SVG's at the point they go into a report and we know what size they should be.  Will you task Maria to clean the SVGs up? @MariaSolovyeva we still have some hours available for you if you can do it this month. Also to clean up is to get the black into a consistent colour  (currently we have grays and blacks of different shades mixed).

**Comment by Charlotte-Morgan** [comment]
Hey @lucernae - great to see the progress on this.  Can I ask where AusAid is coming from in the bottom banner? @felix-yew - is this something you added or in an existing token?

**Comment by felix-yew** [comment]
Hi, mas @lucernae - :+1: this is awesome, glad to see it.

@Charlotte-Morgan - yes, I add this as the supporters title (like in a template that I sent in Asana) or any suggestion for that?

**Comment by Charlotte-Morgan** [comment]
@felix-yew - there are a number of places you can check this. Please take a look at inasafe.org to see how we describe the project supporters. Also look at footers on realtime maps for what is used as a standard. refer to the standard disclaimer text ....
We stopped using AusAID about 2 years ago :)

**Comment by felix-yew** [comment]
@Charlotte-Morgan - Okay, I will revise it. 

![image](https://cloud.githubusercontent.com/assets/13621886/22317610/8182be56-e3a7-11e6-9a88-1fa7d8b9d1c6.png)

Mas @lucernae , fyi: I revise the supporters title.
(Indonesian Government-Australian Government-World Bank-GFDRR)

**Comment by lucernae** [comment]
Ok, noted. I will change in the current PR.

**Comment by Charlotte-Morgan** [comment]
I think this needs to include some acknowledgement of the source of the icons too. 
Icons source: OCHA

**Comment by timlinux** [comment]
@myarjunar can you take this over please

**Comment by Charlotte-Morgan** [comment]
see #3934

**Comment by timlinux** [comment]
We need to make sure the rice and hygiene packs show up in the infographics please.

**Comment by felix-yew** [comment]
Hi @timlinux - I still search the reference of the hygiene packs and additional rice, one of them has been delivered by mas @fredychandra in https://github.com/inasafe/inasafe/issues/3607#issuecomment-285556569 -this ticket has been closed- about additional rice for pregnant and lactating women that come from **Sphere standard**. Maybe we can use it as a reference to the hygiene packs and additional rice.
I create a mock up if we can add the additional rice and hygiene pack.
![inasafe infographic](https://cloud.githubusercontent.com/assets/13621886/24738994/b5905278-1ac3-11e7-85f2-1e45832fbb9b.jpg)

Happy to discuss :smile:

**Comment by Charlotte-Morgan** [comment]
@felix-yew - have you shared the template for this with @myarjunar ?

**Comment by felix-yew** [comment]
This is the new mock up for the infographic
![new infographic v 4](https://cloud.githubusercontent.com/assets/13621886/25834095/e6a600f6-349f-11e7-83b1-98a9a3b7e91c.png)
@myarjunar sorry for the late response.

**Comment by Charlotte-Morgan** [comment]
@felix-yew - did you make a template in QGIS for this?

**Comment by felix-yew** [comment]
>@felix-yew - did you make a template in QGIS for this?

Hi @Charlotte-Morgan - I make this template in power point 😄 because it not used a huge memory when process it. And, I have asked @myarjunar that he work this template in html format. Or, do you want me to create this template with QGIS?

**Comment by Charlotte-Morgan** [comment]
We need to find out from @myarjunar how he makes the template please and if he already has a template we can modify in future?

**Comment by timlinux** [comment]
@felix-yew to design the page layout
@ivanbusthomi will make the template
@myarjunar to implement the layout in code

**Comment by Charlotte-Morgan** [comment]
From DMI side. @felix-yew  will work with DMI to design pages. @ivanbusthomi will work on templates and qpt etc

**Comment by ivanbusthomi** [comment]
Hey @Charlotte-Morgan and @timlinux ,

This is my attempt in designing the qpt template for infographic with atlas feature for each aggregation areas.
![infographic_atlas](https://user-images.githubusercontent.com/7351977/27949653-a71d1106-6328-11e7-8c23-6cdfc200f8c6.gif)

**Comment by Charlotte-Morgan** [comment]
Hi @ivanbusthomi  If you are going to 'atlas' the report for the affected population by location I think the donut will also need to be for the same location ie have some 'atlas' work too

**Comment by ivanbusthomi** [comment]
@Charlotte-Morgan - Yes, the donut should also be customizable if we want to implement atlas for infographic.
I will put the final qpt template for the infographic without the atlas soon after the updating of the design is finished.

**Comment by ivanbusthomi** [comment]
Hi @myarjunar 
this is the qpt template that I create based on @felix-yew templates
* Project and templates
[infographic_templates.zip](https://github.com/inasafe/inasafe/files/1182222/infographic_templates.zip)

* Page 1 
![export_png](https://user-images.githubusercontent.com/7351977/28705086-670e22b4-7398-11e7-97b7-d39a6d8e0088.png)

* Page 2 
![export_png_2](https://user-images.githubusercontent.com/7351977/28705125-8406bd72-7398-11e7-8639-fc60e63cfa0b.png)

**Comment by ivanbusthomi** [comment]
This is some changes in the wording on the template that I mentioned to you earlier.
![infographic_updated_marked](https://user-images.githubusercontent.com/7351977/29120467-7c43d368-7d34-11e7-84ce-3649f22c1d7e.png)
